src/store.js

```diff
@@ -10,14 +11,25 @@ const setProducts = products => ({
 });
 
 export const fetchProducts = () => dispatch => {
-  return axios.get(/* api get route */)
-    .then(products => dispatch(setProducts(products)))
+  return axios.get('api/products/')
+    .then(products => dispatch(setProducts(products.data)))
+}
+
+const addProduct = product => dispatch => {
+  return axios.post('/api/products/', product)
+    .then(() => dispatch(fetchProducts()))
 }
 
-const reducer = (state = {}, action) => {
+const delProduct = id => dispatch => {
+  return axios.delete(`/api/product/${id}`)
+    .then(() => dispatch(fetchProducts()))
+}
+
+const reducer = (state = [], action) => {
   switch (action.type) {
     case SET_PRODUCTS:
-      return state = action.products;
+      state = action.products
```

Comment:
Why not just return action.products? The returned value is set as state, so we don't really need to set the variable state to action.products first.

Reply:
Changed :)